contracts/bindings-perp/src/msg.rs

```diff
@@ -215,6 +215,13 @@ pub enum QueryMsg {
     },
 
     ModuleAccounts {},
+
+    // -----------------------------------------------------------------
+    // From x/oracle
+    // -----------------------------------------------------------------
+    OracleExchangeRate {
+        pair: String,
+    },
```

Comment:
You can use one method to represent several queries here. Let's say that the query here is:

```rust
ExchangeRates {
  pairs: Option<Vec<String>>,
}
```
This would mean you could request a single pair, several, or pass no argument into the request and query all of the pairs. 

The return type could then be:
```rust
use cosmwasm_std::{Decimal};
use cosmwasm_schema::cw_serde;

#[cw_serde]
pub struct ExchangeRatesResponse {
    pub price_map: HashMap<String, Decimal>,
}
```

Using the above approach covers both use cases in an flexible manner while avoiding the need to return errors if certain pairs are missing. If certain pairs don't have exchange rates, the map would simple not have values for those keys.